lib/bridge/MemberListSyncer.js

```diff
@@ -359,16 +389,17 @@ MemberListSyncer.prototype.updateIrcMemberList = Promise.coroutine(function*(cha
             )) {
                 return;
         }
-        this._memberLists.matrix[roomId].remoteJoinedUsers.forEach(
+        let usersToLeave = this._memberLists.matrix[roomId].remoteJoinedUsers.filter(
             (userId) => {
-                if (ircUserIds.indexOf(userId) === -1) {
-                    promises.push(
-                        this.ircBridge.getAppServiceBridge().getIntent(userId).leave(roomId)
-                    );
-                    stats.membership(true, "part");
-                }
+                return ircUserIds.indexOf(userId) === -1;
             }
         );
+        // ID is the channel as we can't use the room ID: we might have 2 channels mapping
+        // to the same room ID.
+        promises.push(this._leaveQueuePool.enqueue(channel, {
```

Comment:
What happens when one channel has 20 users to leave and get's enqueued followed by the same channel with 100 usersToLeave? Will it not just pass back the promise for the first NAMES, leaving only 20 users?

Or maybe we shouldn't care _too_ much about these weird multi-mapping edge cases?